android/app/src/main/java/org/fossasia/openevent/activities/LocationActivity.java

```diff
@@ -101,7 +139,48 @@ protected void onResume() {
         realmRepo.getSessionsByLocation(location)
                 .addChangeListener((sessions, orderedCollectionChangeSet) -> {
                     mSessions.clear();
+                    String upcoming_title="",track="",color;
                     mSessions.addAll(sessions);
+                    sorted_sessions=new ArrayList<Session>();
+                    sorted_sessions.clear();
+                    sorted_sessions.addAll(sessions.sort("startTime"));
+                    DateUtils convert=new DateUtils();
+                    Date current=new Date(117,2,18,16,37);
+                    Date start=new Date();
+                    Date end=new Date();
+                    color = null;
+                    for (Session s:sorted_sessions){
+                        upcoming_title="";
+                        track="";
+                        try {
+                            end=convert.getDate(s.getEndTime());
+                            start=convert.getDate(s.getStartTime());
+                        } catch (ParseException e) {
+                            e.printStackTrace();
+                        }
+                        if ((start.compareTo(current))>0){
```

Comment:
There are date specific methods like `before` and `after`, compareTo is not recommended